gum/backend-darwin/gumdarwinsymbolicator.c

```diff
@@ -457,6 +485,123 @@ gum_darwin_symbolicator_find_functions_matching (GumDarwinSymbolicator * self,
   return (GumAddress *) g_array_free (result, FALSE);
 }
 
+gboolean
+gum_darwin_symbolicator_objc_details_from_address (GumDarwinSymbolicator * self,
+                                                   GumAddress address,
+                                                   GumDebugSymbolDetails * details)
+{
+  gboolean success = FALSE;
+  gchar * selector;
+  GumModuleMap * module_map;
+  const GumModuleDetails * module_details;
+  GumDarwinModule * module;
+  GumCollectFunctionsOperation op;
+  GumCollectedFunction dummy_function, * matched_function;
+  GumApiResolver * resolver;
+
+  selector = NULL;
+  module_map = gum_module_map_new ();
+  module_details = gum_module_map_find (module_map, address);
+
+  if (module_details == NULL)
+    goto unref_module_map;
+
+  module = gum_darwin_module_new_from_memory (module_details->path, self->task,
+      module_details->range->base_address, GUM_DARWIN_MODULE_FLAGS_NONE, NULL);
+
+  op.functions = g_array_new (FALSE, FALSE, sizeof (GumCollectedFunction));
+  op.linkedit = module->image->data;
+
+  gum_darwin_module_enumerate_function_starts (module, gum_collect_functions,
+      &op);
+
+  dummy_function.address = address;
+  dummy_function.size = 0;
+
+  matched_function = bsearch (&dummy_function, op.functions->data,
+      op.functions->len, sizeof (GumCollectedFunction),
+      (GCompareFunc)gum_compare_collected_functions);
+
+  resolver = gum_api_resolver_make ("objc");
```

Comment:
i think `resolver` could be created once lazily then stored in the `GumDarwinSymbolicator` instance (and disposed in the destructor).

The reason is the objc api resolver does a lot of work during creation.